tsdb/wal/wal.go

```diff
@@ -723,40 +724,41 @@ func (w *WAL) fsync(f *Segment) error {
 }
 
 // Close flushes all writes and closes active segment.
-func (w *WAL) Close() (err error) {
+func (w *WAL) Close() error {
 	w.mtx.Lock()
 	defer w.mtx.Unlock()
 
 	if w.closed {
 		return errors.New("wal already closed")
 	}
 
-	if w.segment == nil {
-		w.closed = true
-		return nil
-	}
+	donec := make(chan struct{})
+	w.stopc <- donec
+	<-donec
+
+	var merr tsdb_errors.MultiError
 
 	// Flush the last page and zero out all its remaining size.
 	// We must not flush an empty page as it would falsely signal
 	// the segment is done if we start writing to it again after opening.
 	if w.page.alloc > 0 {
 		if err := w.flushPage(true); err != nil {
-			return err
+			level.Error(w.logger).Log("msg", "flush current page", "err", err)
+			merr.Add(err)
 		}
 	}
 
-	donec := make(chan struct{})
-	w.stopc <- donec
-	<-donec
-
-	if err = w.fsync(w.segment); err != nil {
+	if err := w.fsync(w.segment); err != nil {
 		level.Error(w.logger).Log("msg", "sync previous segment", "err", err)
+		merr.Add(err)
 	}
 	if err := w.segment.Close(); err != nil {
 		level.Error(w.logger).Log("msg", "close previous segment", "err", err)
+		merr.Add(err)
```

Comment:
We did not return these errors before and I am not sure if we want to return them. @bwplotka any opinion here?

Reply:
Ah I did not see that.

I am pretty sure we might want to return but in this case no log needed. (: